internal/services/containers/snapshots_resource.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceSnapshots() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceSnapshotsCreateUpdate,
+		Read:   resourceSnapshotsRead,
+		Update: resourceSnapshotsCreateUpdate,
+		Delete: resourceSnapshotsDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.SnapshotsID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"source_resource_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"snapshot_type": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(containerservice.SnapshotTypeNodePool),
+				}, false),
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceSnapshotsCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Containers.SnapshotsClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := parse.NewSnapshotsID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	t := d.Get("tags").(map[string]interface{})
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id.ResourceGroup, id.SnapshotName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing %s: %s", id, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_snapshots", id.ID())
+		}
+	}
+
+	properties := containerservice.SnapshotProperties{
+		CreationData: &containerservice.CreationData{},
+	}
+
+	if v, ok := d.GetOk("source_resource_id"); ok {
+		properties.CreationData.SourceResourceID = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("snapshot_type"); ok {
+		properties.SnapshotType = containerservice.SnapshotType(v.(string))
+	}
+
+	parameters := containerservice.Snapshot{
+		Location:           &location,
+		SnapshotProperties: &properties,
```

Comment:
Since it's required to configure node pool id, this part can be simplified as 
```
parameters := containerservice.Snapshot{
		Location:           &location,
		SnapshotProperties: &containerservice.CreationData{
SnapshotType : containerservice.SnapshotTypeNodePool,
SourceResourceID: utils.string(d.GetOk("source_resource_id").(string)),
  },
```